File: backend/app/routers/games.py
from fastapi import APIRouter, Depends, HTTPException, Request
from pydantic import BaseModel
from typing import List, Optional, Any
from ..database import get_db
from ..dependencies import verify_admin
import json
import secrets
from datetime import datetime

router = APIRouter()

# --- Pydantic Models ---

class CategoryModel(BaseModel):
    name: str
    slug: str

class PackageModel(BaseModel):
    id: str
    name: str
    price: float
    gameId: str
    subtitle: Optional[str] = None
    description: Optional[str] = None
    badge: Optional[str] = None
    isRecommended: bool
    items: List[str]
    highlights: List[str]
    delivery: Optional[str] = None
    support: Optional[str] = None
    guarantee: Optional[str] = None
    audience: Optional[str] = None
    adminNotes: Optional[str] = None
    isActive: Optional[bool] = None

class GameModel(BaseModel):
    id: str
    name: str
    slug: str
    description: Optional[str] = None
    supportedAndroid: bool
    supportedIos: bool
    warrantyDays: int
    warrantyNote: Optional[str] = None
    isFeatured: bool
    isActive: bool
    referenceTitle: Optional[str] = None
    playImage: Optional[str] = None
    playStore: Optional[str] = None
    catalogType: Optional[str] = None
    category: CategoryModel
    packages: List[PackageModel]
    banStatus: str
    banRiskPercentage: int
    screenshots: Optional[List[str]] = []
    videoUrl: Optional[str] = ""

class GamesResponse(BaseModel):
    success: bool
    data: List[GameModel]

class SaveGameRequest(BaseModel):
    id: Optional[str] = None
    name: str
    slug: Optional[str] = None
    description: str
    category_name: str
    category_slug: str
    supported_android: bool
    supported_ios: bool
    warranty_days: int
    warranty_note: str
    is_featured: bool
    is_active: bool
    play_image: Optional[str] = ""
    banStatus: Optional[str] = "safe"
    banRiskPercentage: Optional[int] = 0
    screenshots: Optional[List[str]] = []
    video_url: Optional[str] = ""

class SavePackageRequest(BaseModel):
    id: Optional[str] = None
    name: str
    game_id: str
    price: float
    subtitle: str
    description: str
    badge: str
    is_recommended: bool
    highlights: List[str]
    delivery: str
    support: str
    guarantee: str
    audience: str
    admin_notes: str
    is_active: bool

# --- Helpers ---

def parse_highlights(highlights_str):
    try:
        return json.loads(highlights_str) if highlights_str else []
    except Exception:
        return []

def package_view(row, is_admin=False):
    payload = {
        "id": row["id"],
        "name": row["name"],
        "price": row["price"],
        "gameId": row["game_id"],
        "subtitle": row["subtitle"],
        "description": row["description"],
        "badge": row["badge"],
        "isRecommended": bool(row["is_recommended"]),
        "items": parse_highlights(row["highlights"]),
        "highlights": parse_highlights(row["highlights"]),
        "delivery": row["delivery"],
        "support": row["support"],
        "guarantee": row["guarantee"],
        "audience": row["audience"]
    }
    if is_admin:
        payload["adminNotes"] = row["admin_notes"]
        payload["isActive"] = bool(row["is_active"])
    return payload

def game_view(row, packages):
    screenshots_raw = row["screenshots"] if "screenshots" in row.keys() else ""
    try:
        screenshots = json.loads(screenshots_raw) if screenshots_raw else []
    except Exception:
        screenshots = []
    
    return {
        "id": row["id"],
        "name": row["name"],
        "slug": row["slug"],
        "description": row["description"],
        "supportedAndroid": bool(row["supported_android"]),
        "supportedIos": bool(row["supported_ios"]),
        "warrantyDays": row["warranty_days"],
        "warrantyNote": row["warranty_note"],
        "isFeatured": bool(row["is_featured"]),
        "isActive": bool(row["is_active"]),
        "referenceTitle": row["reference_title"] if "reference_title" in row.keys() else "",
        "playImage": row["play_image"] if "play_image" in row.keys() else "",
        "playStore": row["play_store"] if "play_store" in row.keys() else "",
        "catalogType": row["catalog_type"] if "catalog_type" in row.keys() else "",
        "category": {"name": row["category_name"], "slug": row["category_slug"]},
        "banStatus": row["ban_status"] if "ban_status" in row.keys() else "safe",
        "banRiskPercentage": row["ban_risk_percentage"] if "ban_risk_percentage" in row.keys() else 0,
        "screenshots": screenshots,
        "videoUrl": row["video_url"] if "video_url" in row.keys() else "",
        "packages": packages
    }

# --- Routes ---

@router.get("/games", response_model=GamesResponse)
def get_public_games(db=Depends(get_db)):
    cursor = db.cursor()
    cursor.execute("SELECT * FROM games WHERE is_active = 1")
    game_rows = cursor.fetchall()
    games_data = []
    for g in game_rows:
        cursor.execute("SELECT * FROM packages WHERE game_id = ? AND is_active = 1", (g["id"],))
        packages = [package_view(p, is_admin=False) for p in cursor.fetchall()]
        games_data.append(game_view(g, packages))
    return {"success": True, "data": games_data}

@router.get("/admin-games", response_model=GamesResponse)
def get_admin_games(admin_user=Depends(verify_admin), db=Depends(get_db)):
    cursor = db.cursor()
    cursor.execute("SELECT * FROM games")
    game_rows = cursor.fetchall()
    games_data = []
    for g in game_rows:
        cursor.execute("SELECT * FROM packages WHERE game_id = ?", (g["id"],))
        packages = [package_view(p, is_admin=True) for p in cursor.fetchall()]
        games_data.append(game_view(g, packages))
    return {"success": True, "data": games_data}

@router.post("/admin-games/{game_id}")
def update_game(game_id: str, game_data: GameModel, db=Depends(verify_admin)):
    return {"success": True}

class ScrapeRequest(BaseModel):
    url: str

@router.post("/admin-scrape-playstore")
def scrape_playstore(req: ScrapeRequest, db=Depends(verify_admin)):
    import urllib.parse
    try:
        from google_play_scraper import app as play_scraper_app
    except ImportError:
        raise HTTPException(status_code=500, detail="google-play-scraper library is not installed.")
        
    url = req.url.strip()
    if not url:
        raise HTTPException(status_code=400, detail="URL is required")
        
    app_id = url
    if "play.google.com" in url and "id=" in url:
        try:
            parsed = urllib.parse.urlparse(url)
            qs = urllib.parse.parse_qs(parsed.query)
            if "id" in qs:
                app_id = qs["id"][0]
        except Exception:
            pass

    try:
        result = play_scraper_app(app_id, lang='th', country='th')
        description = result.get('summary') or result.get('description', '')
        if len(description) > 250:
            description = description[:247] + '...'
            
        genre = result.get('genre', 'Mobile')
        
        screenshots = result.get('screenshots', []) or []
        video_url = result.get('video', '') or ''
        
        return {
            "success": True,
            "data": {
                "name": result.get('title', ''),
                "playImage": result.get('icon', ''),
                "description": description,
                "supportedAndroid": True,
                "playStore": f"https://play.google.com/store/apps/details?id={app_id}",
                "catalogType": genre,
                "screenshots": screenshots[:8],
                "videoUrl": video_url
            }
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Failed to fetch data from Play Store: {str(e)}")

@router.post("/admin-games")
def save_admin_game(req: SaveGameRequest, admin_user=Depends(verify_admin), db=Depends(get_db)):
    cursor = db.cursor()
    game_id = req.id
    if not game_id:
        game_id = f"game-{secrets.token_hex(4)}"
        screenshots_json = json.dumps(req.screenshots or [], ensure_ascii=False)
        cursor.execute("""
            INSERT INTO games (
                id, name, slug, description, category_name, category_slug, 
                supported_android, supported_ios, warranty_days, warranty_note, 
                is_featured, is_active, play_image, ban_status, ban_risk_percentage,
                screenshots, video_url
            ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        """, (
            game_id, req.name, req.slug or req.name.lower().replace(" ", "-"), req.description,
            req.category_name, req.category_slug, int(req.supported_android), int(req.supported_ios),
            req.warranty_days, req.warranty_note, int(req.is_featured), int(req.is_active), req.play_image,
            req.banStatus, req.banRiskPercentage, screenshots_json, req.video_url
        ))
    else:
        screenshots_json = json.dumps(req.screenshots or [], ensure_ascii=False)
        cursor.execute("""
            UPDATE games SET
                name=?, slug=?, description=?, category_name=?, category_slug=?,
                supported_android=?, supported_ios=?, warranty_days=?, warranty_note=?,
                is_featured=?, is_active=?, play_image=?, ban_status=?, ban_risk_percentage=?,
                screenshots=?, video_url=?
            WHERE id=?
        """, (
            req.name, req.slug, req.description, req.category_name, req.category_slug,
            int(req.supported_android), int(req.supported_ios), req.warranty_days, req.warranty_note,
            int(req.is_featured), int(req.is_active), req.play_image, req.banStatus, req.banRiskPercentage,
            screenshots_json, req.video_url, game_id
        ))
    db.commit()
    return {"success": True}

@router.post("/admin-packages")
def save_admin_package(req: SavePackageRequest, admin_user=Depends(verify_admin), db=Depends(get_db)):
    cursor = db.cursor()
    pkg_id = req.id
    highlights_json = json.dumps(req.highlights, ensure_ascii=False)
    
    if not pkg_id:
        pkg_id = f"pkg-{secrets.token_hex(4)}"
        cursor.execute("""
            INSERT INTO packages (
                id, game_id, name, price, subtitle, description, badge,
                is_recommended, highlights, delivery, support, guarantee,
                audience, admin_notes, is_active
            ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        """, (
            pkg_id, req.game_id, req.name, req.price, req.subtitle, req.description,
            req.badge, int(req.is_recommended), highlights_json, req.delivery,
            req.support, req.guarantee, req.audience, req.admin_notes, int(req.is_active)
        ))
    else:
        cursor.execute("""
            UPDATE packages SET
                name=?, price=?, subtitle=?, description=?, badge=?,
                is_recommended=?, highlights=?, delivery=?, support=?,
                guarantee=?, audience=?, admin_notes=?, is_active=?
            WHERE id=?
        """, (
            req.name, req.price, req.subtitle, req.description, req.badge,
            int(req.is_recommended), highlights_json, req.delivery, req.support,
            req.guarantee, req.audience, req.admin_notes, int(req.is_active), pkg_id
        ))
    db.commit()
    return {"success": True}

File: backend-node/src/routes/games.js
const express = require('express');
const router = express.Router();
const crypto = require('crypto');
const db = require('../db/helpers');
const { verifyAdmin } = require('../middleware/auth');

// Helper to parse JSON strings from DB safely
function parseJsonArray(str) {
  if (!str) return [];
  try {
    // If it's already an array, return it
    if (Array.isArray(str)) return str;
    return JSON.parse(str);
  } catch (e) {
    return [];
  }
}

// Map database row to package format
function packageView(row, isAdmin = false) {
  const highlights = parseJsonArray(row.highlights);
  const data = {
    id: row.id,
    name: row.name,
    price: row.price,
    gameId: row.game_id,
    subtitle: row.subtitle || '',
    description: row.description || '',
    badge: row.badge || '',
    isRecommended: Boolean(row.is_recommended),
    items: highlights,
    highlights: highlights,
    delivery: row.delivery || 'Instant',
    support: row.support || '24/7',
    guarantee: row.guarantee || '100% Safe',
    audience: row.audience || ''
  };

  if (isAdmin) {
    data.adminNotes = row.admin_notes || '';
    data.isActive = Boolean(row.is_active);
  }
  return data;
}

// Map database row to game format
function gameView(row, packages) {
  return {
    id: row.id,
    name: row.name,
    slug: row.slug,
    description: row.description || '',
    supportedAndroid: Boolean(row.supported_android),
    supportedIos: Boolean(row.supported_ios),
    warrantyDays: row.warranty_days || 0,
    warrantyNote: row.warranty_note || '',
    isFeatured: Boolean(row.is_featured),
    isActive: Boolean(row.is_active),
    referenceTitle: row.reference_title || '',
    playImage: row.play_image || '',
    playStore: row.play_store || '',
    catalogType: row.catalog_type || 'default',
    category: {
      name: row.category_name || '',
      slug: row.category_slug || ''
    },
    banStatus: row.ban_status || 'safe',
    banRiskPercentage: row.ban_risk_percentage || 0,
    screenshots: parseJsonArray(row.screenshots),
    videoUrl: row.video_url || '',
    packages
  };
}

// GET /api/games
router.get('/games', async (req, res) => {
  try {
    const gameRows = await db.all('SELECT * FROM games WHERE is_active = 1');
    const gamesData = [];

    for (const g of gameRows) {
      const pkgRows = await db.all('SELECT * FROM packages WHERE game_id = ? AND is_active = 1', [g.id]);
      const packages = pkgRows.map(p => packageView(p, false));
      gamesData.push(gameView(g, packages));
    }

    return res.status(200).json({ success: true, data: gamesData });
  } catch (error) {
    console.error('Get public games error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// GET /api/games/:id
router.get('/games/:id', async (req, res) => {
  try {
    const { id } = req.params;
    const g = await db.get('SELECT * FROM games WHERE id = ? OR slug = ?', [id, id]);
    if (!g) {
      return res.status(404).json({ success: false, error: 'Game not found' });
    }

    const pkgRows = await db.all('SELECT * FROM packages WHERE game_id = ? AND is_active = 1', [g.id]);
    const packages = pkgRows.map(p => packageView(p, false));

    return res.status(200).json({ success: true, data: gameView(g, packages) });
  } catch (error) {
    console.error('Get single game error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// GET /api/admin-games
router.get('/admin-games', verifyAdmin, async (req, res) => {
  try {
    const gameRows = await db.all('SELECT * FROM games');
    const gamesData = [];

    for (const g of gameRows) {
      const pkgRows = await db.all('SELECT * FROM packages WHERE game_id = ?', [g.id]);
      const packages = pkgRows.map(p => packageView(p, true));
      gamesData.push(gameView(g, packages));
    }

    return res.status(200).json({ success: true, data: gamesData });
  } catch (error) {
    console.error('Get admin games error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// POST /api/admin-scrape-playstore
router.post('/admin-scrape-playstore', verifyAdmin, (req, res) => {
  try {
    const { url } = req.body;
    if (!url) {
      return res.status(400).json({ success: false, error: 'URL is required' });
    }

    let appId = url;
    if (url.includes('play.google.com') && url.includes('id=')) {
      try {
        const u = new URL(url);
        appId = u.searchParams.get('id') || url;
      } catch (e) {}
    }

    // Return a realistic mock response matching python scraper return format
    return res.status(200).json({
      success: true,
      data: {
        name: 'Idle Hero TD',
        playImage: 'https://play-lh.googleusercontent.com/icon',
        description: 'A mock description of Idle Hero TD.',
        supportedAndroid: true,
        playStore: `https://play.google.com/store/apps/details?id=${appId}`,
        catalogType: 'Strategy',
        screenshots: [
          'https://play-lh.googleusercontent.com/screenshot1',
          'https://play-lh.googleusercontent.com/screenshot2'
        ],
        videoUrl: 'https://www.youtube.com/watch?v=mock'
      }
    });
  } catch (error) {
    console.error('Scrape playstore error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// POST /api/admin-games
router.post('/admin-games', verifyAdmin, async (req, res) => {
  try {
    const {
      id, name, slug, description, category_name, category_slug,
      supported_android, supported_ios, warranty_days, warranty_note,
      is_featured, is_active, play_image, banStatus, banRiskPercentage,
      screenshots, video_url, supportedAndroid, supportedIos,
      warrantyDays, warrantyNote, isFeatured, isActive, playImage,
      videoUrl
    } = req.body;

    if (!name) {
      return res.status(400).json({ success: false, error: 'Name is required' });
    }

    const finalSlug = slug || name.toLowerCase().replace(/ /g, '-').replace(/[^\w-]/g, '');
    const finalCategoryName = category_name || '';
    const finalCategorySlug = category_slug || '';
    
    // Resolve camelCase and snake_case values
    const finalSupportedAndroid = supported_android !== undefined ? (supported_android ? 1 : 0) : (supportedAndroid !== undefined ? (supportedAndroid ? 1 : 0) : 1);
    const finalSupportedIos = supported_ios !== undefined ? (supported_ios ? 1 : 0) : (supportedIos !== undefined ? (supportedIos ? 1 : 0) : 1);
    const finalWarrantyDays = warranty_days !== undefined ? warranty_days : (warrantyDays !== undefined ? warrantyDays : 0);
    const finalWarrantyNote = warranty_note !== undefined ? warranty_note : (warrantyNote !== undefined ? warrantyNote : '');
    const finalIsFeatured = is_featured !== undefined ? (is_featured ? 1 : 0) : (isFeatured !== undefined ? (isFeatured ? 1 : 0) : 0);
    const finalIsActive = is_active !== undefined ? (is_active ? 1 : 0) : (isActive !== undefined ? (isActive ? 1 : 0) : 1);
    
    const finalPlayImage = play_image !== undefined ? play_image : (playImage !== undefined ? playImage : '');
    const finalBanStatus = banStatus !== undefined ? banStatus : 'safe';
    const finalBanRisk = banRiskPercentage !== undefined ? banRiskPercentage : 0;
    const finalScreenshots = JSON.stringify(screenshots || []);
    const finalVideoUrl = video_url !== undefined ? video_url : (videoUrl !== undefined ? videoUrl : '');

    let gameId = id;
    if (!gameId) {
      gameId = 'game-' + crypto.randomBytes(4).toString('hex');
      await db.run(
        `INSERT INTO games (
          id, name, slug, description, category_name, category_slug, 
          supported_android, supported_ios, warranty_days, warranty_note, 
          is_featured, is_active, play_image, ban_status, ban_risk_percentage,
          screenshots, video_url
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
        [
          gameId, name, finalSlug, description || '', finalCategoryName, finalCategorySlug,
          finalSupportedAndroid, finalSupportedIos, finalWarrantyDays, finalWarrantyNote,
          finalIsFeatured, finalIsActive, finalPlayImage, finalBanStatus, finalBanRisk,
          finalScreenshots, finalVideoUrl
        ]
      );
    } else {
      // Check if game exists
      const existing = await db.get('SELECT id FROM games WHERE id = ?', [gameId]);
      if (!existing) {
        return res.status(404).json({ success: false, error: 'Game not found' });
      }

      await db.run(
        `UPDATE games SET
          name=?, slug=?, description=?, category_name=?, category_slug=?,
          supported_android=?, supported_ios=?, warranty_days=?, warranty_note=?,
          is_featured=?, is_active=?, play_image=?, ban_status=?, ban_risk_percentage=?,
          screenshots=?, video_url=?
        WHERE id=?`,
        [
          name, finalSlug, description || '', finalCategoryName, finalCategorySlug,
          finalSupportedAndroid, finalSupportedIos, finalWarrantyDays, finalWarrantyNote,
          finalIsFeatured, finalIsActive, finalPlayImage, finalBanStatus, finalBanRisk,
          finalScreenshots, finalVideoUrl, gameId
        ]
      );
    }

    return res.status(200).json({ success: true, message: 'Game saved successfully' });
  } catch (error) {
    console.error('Save game error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// POST /api/admin-games/:game_id
router.post('/admin-games/:game_id', verifyAdmin, async (req, res) => {
  try {
    const gameId = req.params.game_id;
    const {
      name, slug, description, category_name, category_slug,
      supported_android, supported_ios, warranty_days, warranty_note,
      is_featured, is_active, play_image, banStatus, banRiskPercentage,
      screenshots, video_url, supportedAndroid, supportedIos,
      warrantyDays, warrantyNote, isFeatured, isActive, playImage,
      playStore, catalogType, ban_status, ban_risk_percentage, videoUrl
    } = req.body;

    const game = await db.get('SELECT id FROM games WHERE id = ?', [gameId]);
    if (!game) {
      return res.status(404).json({ success: false, error: 'Game not found' });
    }

    const finalName = name !== undefined ? name : '';
    const finalSlug = slug || (name ? name.toLowerCase().replace(/ /g, '-') : '');
    const finalDescription = description || '';
    const finalCategoryName = category_name || '';
    const finalCategorySlug = category_slug || '';
    
    const finalSupportedAndroid = supported_android !== undefined ? (supported_android ? 1 : 0) : (supportedAndroid !== undefined ? (supportedAndroid ? 1 : 0) : 1);
    const finalSupportedIos = supported_ios !== undefined ? (supported_ios ? 1 : 0) : (supportedIos !== undefined ? (supportedIos ? 1 : 0) : 1);
    const finalWarrantyDays = warranty_days !== undefined ? warranty_days : (warrantyDays !== undefined ? warrantyDays : 0);
    const finalWarrantyNote = warranty_note !== undefined ? warranty_note : (warrantyNote !== undefined ? warrantyNote : '');
    const finalIsFeatured = is_featured !== undefined ? (is_featured ? 1 : 0) : (isFeatured !== undefined ? (isFeatured ? 1 : 0) : 0);
    const finalIsActive = is_active !== undefined ? (is_active ? 1 : 0) : (isActive !== undefined ? (isActive ? 1 : 0) : 1);
    
    const finalPlayImage = play_image !== undefined ? play_image : (playImage !== undefined ? playImage : '');
    const finalBanStatus = banStatus !== undefined ? banStatus : (ban_status !== undefined ? ban_status : 'safe');
    const finalBanRisk = banRiskPercentage !== undefined ? banRiskPercentage : (ban_risk_percentage !== undefined ? ban_risk_percentage : 0);
    const finalScreenshots = JSON.stringify(screenshots || []);
    const finalVideoUrl = video_url !== undefined ? video_url : (videoUrl !== undefined ? videoUrl : '');
    const finalPlayStore = playStore || '';
    const finalCatalogType = catalogType || '';

    await db.run(
      `UPDATE games SET
        name = ?, slug = ?, description = ?, category_name = ?, category_slug = ?,
        supported_android = ?, supported_ios = ?, warranty_days = ?, warranty_note = ?,
        is_featured = ?, is_active = ?, play_image = ?, ban_status = ?, ban_risk_percentage = ?,
        screenshots = ?, video_url = ?, play_store = ?, catalog_type = ?
      WHERE id = ?`,
      [
        finalName, finalSlug, finalDescription, finalCategoryName, finalCategorySlug,
        finalSupportedAndroid, finalSupportedIos, finalWarrantyDays, finalWarrantyNote,
        finalIsFeatured, finalIsActive, finalPlayImage, finalBanStatus, finalBanRisk,
        finalScreenshots, finalVideoUrl, finalPlayStore, finalCatalogType, gameId
      ]
    );

    return res.status(200).json({ success: true, message: 'Game updated successfully' });
  } catch (error) {
    console.error('Update game error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// POST /api/admin-packages
router.post('/admin-packages', verifyAdmin, async (req, res) => {
  try {
    const {
      id, game_id, name, price, subtitle, description, badge,
      is_recommended, highlights, delivery, support, guarantee,
      audience, admin_notes, is_active
    } = req.body;

    if (!name || !game_id || price === undefined) {
      return res.status(400).json({ success: false, error: 'Required: name, game_id, price' });
    }

    const highlightsJson = JSON.stringify(highlights || []);
    const finalDelivery = delivery || 'Instant';
    const finalSupport = support || '24/7';
    const finalGuarantee = guarantee || '100% Safe';
    const finalIsActive = is_active !== undefined ? (is_active ? 1 : 0) : 1;
    const finalIsRecommended = is_recommended !== undefined ? (is_recommended ? 1 : 0) : 0;

    let pkgId = id;
    if (!pkgId) {
      pkgId = 'pkg-' + crypto.randomBytes(4).toString('hex');
      await db.run(
        `INSERT INTO packages (
          id, game_id, name, price, subtitle, description, badge,
          is_recommended, highlights, delivery, support, guarantee,
          audience, admin_notes, is_active
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
        [
          pkgId, game_id, name, price, subtitle || '', description || '', badge || '',
          finalIsRecommended, highlightsJson, finalDelivery, finalSupport, finalGuarantee,
          audience || '', admin_notes || '', finalIsActive
        ]
      );
    } else {
      await db.run(
        `UPDATE packages SET
          name=?, price=?, subtitle=?, description=?, badge=?,
          is_recommended=?, highlights=?, delivery=?, support=?,
          guarantee=?, audience=?, admin_notes=?, is_active=?
        WHERE id=?`,
        [
          name, price, subtitle || '', description || '', badge || '',
          finalIsRecommended, highlightsJson, finalDelivery, finalSupport,
          finalGuarantee, audience || '', admin_notes || '', finalIsActive, pkgId
        ]
      );
    }

    return res.status(200).json({ success: true, message: 'Package saved successfully' });
  } catch (error) {
    console.error('Save package error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// DELETE /api/admin-games/:id
router.delete('/admin-games/:id', verifyAdmin, async (req, res) => {
  try {
    const { id } = req.params;
    await db.run('DELETE FROM games WHERE id = ?', [id]);
    await db.run('DELETE FROM packages WHERE game_id = ?', [id]);
    return res.status(200).json({ success: true, message: 'Game deleted successfully' });
  } catch (error) {
    console.error('Delete game error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

// POST /api/admin-games/delete
router.post('/admin-games/delete', verifyAdmin, async (req, res) => {
  try {
    const { game_id } = req.body;
    if (!game_id) {
      return res.status(400).json({ success: false, error: 'game_id is required' });
    }
    await db.run('DELETE FROM games WHERE id = ?', [game_id]);
    await db.run('DELETE FROM packages WHERE game_id = ?', [game_id]);
    return res.status(200).json({ success: true, message: 'Game deleted successfully' });
  } catch (error) {
    console.error('Delete game POST error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

module.exports = router;
